tutorials/high_performance_networking/README.md

```diff
@@ -0,0 +1,2171 @@
+    |                                                                                        \-00.1  NVIDIA Corporation GA102 High Definition Audio Controller
+    +-[0004:00]---00.0-[01-ff]----00.0  Sandisk Corp WD PC SN810 / Black SN850 NVMe SSD
+    +-[0001:00]---00.0-[01-ff]----00.0-[02-fc]--+-01.0-[03-34]----00.0  Realtek Semiconductor Co., Ltd. RTL8111/8168/8411 PCI Express Gigabit Ethernet Controller
+    |                                           +-02.0-[35-66]----00.0  Realtek Semiconductor Co., Ltd. RTL8111/8168/8411 PCI Express Gigabit Ethernet Controller
+    |                                           +-03.0-[67-98]----00.0  Device 1c00:3450
+    |                                           +-04.0-[99-ca]----00.0-[9a]--+-00.0  ASPEED Technology, Inc. ASPEED Graphics Family
+    |                                           |                            \-02.0  ASPEED Technology, Inc. Device 2603
+    |                                           \-05.0-[cb-fc]----00.0  Realtek Semiconductor Co., Ltd. RTL8822CE 802.11ac PCIe Wireless Network Adapter
+    \-[0000:00]-
+    ```
+
+!!! warning "x86_64 compatibility"
+
+    Most x86_64 systems are not designed for this topology as they lack a discrete PCIe switch. In that case, the best connection they can achieve is `NODE`.
+
+
+### 3.2 Check the NIC's PCIe configuration
+
+??? example "WIP"
+
+    See <https://enterprise-support.nvidia.com/s/article/understanding-pcie-configuration-for-maximum-performance>
+
+=== "tune_system.py"
+
+    === "Debian installation"
+
+        ```bash
+        sudo /opt/nvidia/holoscan/bin/tune_system.py --check mps
+        ```
+
+    === "HoloHub sources"
+
+        ```bash
+        cd holohub
+        sudo ./operators/advanced_network/python/tune_system.py --check mps
+        ```
+
+=== "manual"
+
+    Identify the PCIe address of your NVIDIA NIC:
+
+    === "ibdev2netdev"
+
+        ```bash
+        nic_pci=$(sudo ibdev2netdev -v | awk '{print $1}' | head -n1)
+        ```
+
+    === "lspci"
+
+        ```bash
+        # `0200` is the PCI-SIG class code for NICs
+        # `15b3` is the Vendor ID for Mellanox
+        nic_pci=$(lspci -n | awk '$2 == "0200:" && $3 ~ /^15b3:/ {print $1}' | head -n1)
+        ```
+
+    Check current MPS:
+
+    ```bash
+    sudo lspci -vv -s $bus_addr | grep DevCtl: -A2 | grep -oE "MaxPayload [0-9]+"
+    ```
+
+    And max MPS:
+
+    ```bash
+    sudo lspci -vv -s $bus_addr | grep DevCap: -A1 | grep -oE "MaxPayload [0-9]+"
+    ```
+
+### 3.3 Maximize the NIC's Max Read Request Size (MRRS)
+
+??? example "WIP"
+
+    See <https://enterprise-support.nvidia.com/s/article/understanding-pcie-configuration-for-maximum-performance>
+
+Check current MRRS:
+
+=== "tune_system.py"
+
+    === "Debian installation"
+
+        ```bash
+        sudo /opt/nvidia/holoscan/bin/tune_system.py --check mrrs
```

Comment:
Should we mention here that this value is reset on reboot and needs to be set every time the system boots?

Reply:
oh :) yes. I guess in the future we could add a flag to the setters for persistence through systemd.